### PR TITLE
lower fielddata circuit breaker's default limit

### PR DESCRIPTION
**Pull request by PnPie**
As discussed in #27130, change fielddata circuit breaker default limit from 60% to 30%.

### CODE REVIEW

**Comment by dnhatn** [comment]
@elasticmachine please test this.

**Comment by dakrone** [comment]
In the #27130 we discussed going to 40% though, not 30%?

**Comment by PnPie** [comment]
I changed it to 40%.

**Comment by danielmitterdorfer** [comment]
@PnPie I had a quick look at your PR. Could you please:

1. Merge master and fix merge conflicts
2. Update the documentation in [circuit_breaker.asciidoc](https://github.com/elastic/elasticsearch/blob/master/docs/reference/modules/indices/circuit_breaker.asciidoc#field-data-circuit-breaker) so it reflects the new default.
3. Add a short note about this change to the [breaking changes docs for 7.0](https://github.com/elastic/elasticsearch/blob/master/docs/reference/migration/migrate_7_0/indices.asciidoc) next to "Parent circuit breaker changes"?

**Comment by PnPie** [comment]
Hi @danielmitterdorfer I updated the PR with the changes you mentioned.

**Comment by danielmitterdorfer** [comment]
@elasticmachine test this please.

**Comment by danielmitterdorfer** [comment]
@elasticmachine test this please.

File: docs/reference/migration/migrate_7_0/indices.asciidoc [hunk 1]
```diff
@@ -92,6 +92,11 @@ heap memory instead of only considering the reserved memory by child circuit bre
 setting is `true`, the default parent breaker limit also changes from 70% to 95% of the JVM heap size.
 The previous behavior can be restored by setting `indices.breaker.total.use_real_memory` to `false`.
 
+[float]
+==== Field data circuit breaker changes
+Lower the default field data circuit breaker limit from 60% to 40% as `doc_values` was introduced for
```
Comment (danielmitterdorfer) [inline docs/reference/migration/migrate_7_0/indices.asciidoc]: Can we please change the wording a little bit?

```suggestion
As doc values have been enabled by default in earlier versions of Elasticsearch,
there is less need for fielddata. Therefore, the default value of the setting
`indices.breaker.fielddata.limit` has been lowered from 60% to 40% of the JVM
heap size. 
```

**Review by danielmitterdorfer — CHANGES_REQUESTED** [review]
@PnPie thanks for following up! Overall the change looks fine and also CI was green. I'd just ask you to change the wording in the migration docs (I've left a suggestion). Once this is in, I think we're good to merge. :)

**Comment by PnPie** [comment]
Hi @danielmitterdorfer I changed the docs as you suggested.

**Comment by danielmitterdorfer** [comment]
@elasticmachine test this please.

**Review by danielmitterdorfer — APPROVED** [review]
Looks good to me. Thank you for your pull request and thanks for iterating @PnPie. Once the CI build is green, I'll merge it.

**Comment by danielmitterdorfer** [comment]
@elasticmachine test this please.

**Comment by danielmitterdorfer** [comment]
This closes #27130

**Pull request merged**